### ISSUE TITLE
How to add more parameters into batch

### ISSUE THREAD

**Issue by anguoyang** [PyTorchLightning/pytorch-lightning#1888]
in the call back function:
```
def training_step(self, batch, batch_idx):
        x, y = batch
        y_logits = self.forward(x)
```

sometimes  we need more parameters besides x and y, for example in face detection, we need 3: x, priors, y, the priors is the anchor boxes, we need to do additional operations between  y_logits and priors. like this:
```
def training_step(self, batch, batch_idx):
        x, priors, y = batch
        y_logits = self.forward(x)
        out = otherOperation(priors, y_logits)
```
How could we pass the priors into batch?

**Comment by HansBambel** [comment]
Depending on what you return in your Dataset `__getitem__` you can get multiple values back.

If your Dataset has `return img, target` you do `x, y = batch`. 
If it has `return img, mask, target` you do `x, mask, y = batch`.
